client/client-multi/client-cpp/impl/notification/NotificationTransport.cpp

```diff
@@ -29,33 +29,25 @@
 namespace kaa {
 
 NotificationTransport::NotificationTransport(IKaaClientStateStoragePtr status, IKaaChannelManager& manager)
-    : AbstractKaaTransport(manager), notificationProcessor_(nullptr)
+                                           : AbstractKaaTransport(manager), notificationProcessor_(nullptr), topicListHash_(0)
 {
     setClientState(status);
-
-    const DetailedTopicStates& detailedStatesContainer = clientStatus_->getTopicStates();
-    for (const auto& state : detailedStatesContainer) {
-        notificationSubscriptions_.insert(std::make_pair(state.second.topicId, state.second.sequenceNumber));
-    }
 }
 
 NotificationSyncRequestPtr NotificationTransport::createEmptyNotificationRequest()
 {
     NotificationSyncRequestPtr request(new NotificationSyncRequest);
 
-    request->appStateSeqNumber = clientStatus_->getNotificationSequenceNumber();
-
-    /*TODO: topic list hash is a future feature */
-    request->topicListHash.set_null();
+    request->topicListHash = clientStatus_->getTopicListHash();
 
-    const DetailedTopicStates& detailedStatesContainer = clientStatus_->getTopicStates();
-    if (!detailedStatesContainer.empty()) {
-        std::vector<TopicState> container(detailedStatesContainer.size());
-        auto it = detailedStatesContainer.begin();
+	const auto& topicStates = clientStatus_->getTopicStates();
+    if (!topicStates.empty()) {
```

Comment:
This code snippet duplicates both in createNotificationRequest() and in createEmptyNotificationRequest().
Could you move it to a separate function?